coretex/entities/dataset/network_dataset.py

```diff
@@ -224,6 +240,14 @@ def createCacheDataset(cls, prefix: str, dependencies: List[str], projectId: int
 
         return dataset
 
+    def finalizeState(self) -> None:
+        parameters = {
+            "name": self.name,
+            "state": DatasetState.final
+        }
+
+        networkManager.put(f"dataset/:{self.id}", parameters)
```

Comment:
Instead of `networkManager.put` use `NetworkDataset.update` method to do this. Handle response of the request appropriatelly.